[Avro] @AvroNamespace annotation to override Avro type namespace

MichalFoksa opened the pull request:
New `AvroNamespace` annotation to override default Avro type namespace.

Simple way how to override default Avro namespace value with a new, custom, annotation.

This solves #310.

Is the new annotation package OK?
```
com.fasterxml.jackson.dataformat.avro.AvroFixedSize              -  the only existing annotation
com.fasterxml.jackson.dataformat.avro.annotations.AvroNamespace  - new annotation
```

MichalFoksa commented:
New `AvroNamespace` annotation to ovveride default Avro type namespace.

I think I found simple way how to evverige default Avro namespace vallue with a new, custom, annotation.

This solves #310.

Is the new annotation package OK?
```
com.fasterxml.jackson.dataformat.avro.AvroFixedSize              -  the only existing annotation
com.fasterxml.jackson.dataformat.avro.annotations.AvroNamespace  - new annotation
```

cowtowncoder commented on avro/src/main/java/com/fasterxml/jackson/dataformat/avro/annotations/AvroNamespace.java:
```diff
@@ -0,0 +1,16 @@
+ * Annotation allows to override default Avro type namespace value.
+ * Default value is Java package name.
+ */
+@Target(ElementType.TYPE)
```
To allow "annotation bundle" usage may also want to add `ElementType.ANNOTATION_TYPE`

MichalFoksa replied:
Right, good idea. 
ElementType.ANNOTATION_TYPE added to allow "annotation bundle" usage.

cowtowncoder commented:
I think I'll let Amazon Ion folks decide on general applicability but one thing is that this cannot go in 2.13 patch since it is both slightly backwards-incompatible (changes method signature of the helper method) and (more importantly) adds a new annotation type.
So please re-base against 2.14.

MichalFoksa commented:
PR is now rebased on  2.14.

cowtowncoder commented:
Thank you @MichalFoksa! 

Also, apologies for nonsensical comment wrt Ion & Amazon developers. This is for Avro, but somehow I managed to get things fixed.

I'll try to get this reviewed soon,

MichalFoksa commented:
@cowtowncoder 

No problem, really - I understand :).

I have updated my commit - just reorganized imports in AvroSchemaHelper.  Moved imports from `com.fasterxml.jackson` together.

cowtowncoder commented:
Will merge: will also change package name from "...avro.annotations" to "...avro.annotation", just to be consistent with other Jackson packages (jackson-annotations, jackson-databind, jackson-dataformat-xml).

cowtowncoder commented:
Thank you again @MichalFoksa for contributing this! It will go in 2.14.0 once we get there.
